lib/strings.nix

```diff
@@ -678,4 +678,36 @@ rec {
        => "1.0"
   */
   fileContents = file: removeSuffix "\n" (builtins.readFile file);
+
+
+  /* Creates a valid derivation name from a potentially invalid one.
+
+     Type: sanitizeDerivationName :: String -> String
+
+     Example:
+       sanitizeDerivationName "../hello.bar # foo"
+       => "-hello.bar-foo"
+       sanitizeDerivationName ""
+       => "unknown"
+       sanitizeDerivationName pkgs.hello
+       => "-nix-store-2g75chlbpxlrqn15zlby2dfh8hr9qwbk-hello-2.10"
+  */
+  sanitizeDerivationName = lib.flip lib.pipe [
```

Comment:
```suggestion
  sanitizeDerivationName = string: lib.pipe string [
```

Reply:
(noob question about how this works!)

Does this remove the dependency on `lib.flip` by explicitly accepting the `string` as an argument before passing it on to `pipe`'s first argument? In essence, this sacrifices the point-free solution for 1 less dependency?

Reply:
It's less about reducing dependencies, but rather reducing mental overhead. By not using `flip` one doesn't need to know what `flip` does to understand this function.